slamd/design_assistant/processing/llm_service.py

```diff
@@ -44,34 +37,88 @@ def use_free_tier_token(cls):
+        user_excerpt = material_type_excerpt + '////Components: \n' + powders_excerpt + liquid_excerpt + other_excerpt + comment_excerpt + design_targets_excerpt
         # combine user prompt with system prompt and instruction to build final prompt
-        system_prompt = "You have performed thousands of experiments in the laboratory. You have extensive design proficiency in the compressive strength of FA/GGBFS-based geopolymer concrete. You can answer questions succinctly because you know that each question relates to only one part of the big picture. You always answer with no more than 8 concise sentences, each containing quantitative facts and trade-offs that relate only to the compressive strength. You always answer directly, e.g., the change in (parameter) between (lower) and (upper) range has (effect) due to (influence).  Let’s work this out in a step-by-step way to be sure we have the right answer. Consider the following scenario:\n";
-        instruction = f'What is the best design knowledge you have for finding concrete formulations, that consist of the specified components, adhere to the specified design targets and have the highest possible compressive strength?'
-        prompt = system_prompt + user_prompt + instruction
-        return prompt
+        system_excerpt = cls._generate_design_knowledge_system_excerpt(zero_shot_learner_session)
+        material_type = zero_shot_learner_session['type']
+        instruction_excerpt = f'What is the best design knowledge you have for finding {material_type} formulations, that consist of the specified components, adhere to the specified design targets and have the highest possible compressive strength?'
+        design_knowledge_prompt = system_excerpt + user_excerpt + instruction_excerpt
+        return design_knowledge_prompt
+
+    @classmethod
+    def _generate_design_knowledge_system_excerpt(cls, zero_shot_learner_session):
+        design_targets = zero_shot_learner_session['design_targets']
+        design_targets_names = [design_target["design_target_name_field"] for design_target in design_targets]
+        if len(design_targets_names) > 1:
+            design_targets_names = ' and '.join(design_targets_names)
+        system_excerpt =  f"You have performed thousands of experiments in the laboratory. You have extensive design proficiency in the compressive strength of cementitious materials. You can answer questions succinctly because you know that each question relates to only one part of the big picture. You always answer with no more than 8 concise sentences, each containing quantitative facts and trade-offs that relate only to the {design_targets_names}. You always answer directly, e.g., the change in (parameter) between (lower) and (upper) range has (effect) due to (influence).  Let’s work this out in a step-by-step way to be sure we have the right answer. Consider the following scenario:\n";
```

Comment:
Nur proficiency in the compressive strength?? Was ist mit anderen Design targets? Vllt eher so was wie proficiency properties of cementitious...